src/OpenTelemetry/Metrics/AggregationTemporality.cs

```diff
@@ -30,15 +30,5 @@ public enum AggregationTemporality : byte
         /// Delta.
         /// </summary>
         Delta = 0b10,
-
```

Comment:
Follow up on https://github.com/open-telemetry/opentelemetry-dotnet/pull/2353#discussion_r709361948.

I struggled between `AggregationTemporality` and `AggregationTemporalityModes`, it seems `AggregationTemporality` is a well established term in OpenTelemetry (IMHO), and `AggregationTemporalityModes` just looks ... weird to me.

@CodeBlanch @cijothomas @alanwest do you feel strong about this?

Reply:
`AggregationTemporality` sounds good.. Yes  "modes" is bit weird

Reply:
Yeah, I think `ExportModes` looks nice because `Export` is a noun.
I also struggled with `AggregationModes` for few seconds and concluded it was a bad idea.

Reply:
I'm OK with keeping `AggregationTemporality`! I don't have an issue with `AggregationModes.Delta` either though. To me, being much further away from the metrics discussions, `AggregationModes.Delta` is immediately clear at a glance as to its purpose. "AggregationTemporality" I would need to google "temporality" to understand it. Here's what Google gave me: _the state of existing within or having some relationship with time_ 🧙‍♀️ 

But we could also clarify with XML comments.

Reply:
Trying to be a bit creative here, how about:
* `AggregationModes.CumulativeTemporality`
* `AggregationModes.DeltaTemporality`

Reply:
> Here's what Google gave me: _the state of existing within or having some relationship with time 🧙‍♀️_

😆 

Completely agree that the first time hearing "aggregation temporality" is definitely a "what did you just say to me?" moment... that said, to @reyang's point, it has become an established term in OpenTelemetry, so I lean more towards keeping `AggregationTemporality` in the name.

I'm a bit skeptical that `ExplicitReset`/`ImplicitReset` would ever be a thing, but I am interested if there's been any discussions about this possibility.

Reply:
```
AggregationModes.CumulativeTemporality
AggregationModes.DeltaTemporality
```

sounds good to me.

Reply:
We have 3 votes for `AggregationModes` and @alanwest prefers `AggregationTemporality`.
I will keep `AggregationTemporality` in this PR, once we merged it, I'll create a separate PR to rename it to `AggregationModes` and we can discuss more there.

Reply:
Tracked by #2371.